### PR TITLE
Pipebombs now require ignition source

### LINKED ISSUE #26239: Pipe bombs should require to be physically lit
## Description
Either by lighter, welder, getting set on fire, or the area being on fire (with a flammable gas). The crafting recipe has no igniter, and it literally has an unlit fuse on the sprite. Also very cool and cinematic physically lighting an explosive.
h

**Additional context**
Maybe you could attatch an igniter to make a remote pipe bomb?

### PR DESCRIPTION
**Pull request by ChaseFlorom**
## About the PR

Pipebombs must now be ignited by things like lighters/welders.

## Why / Balance

It makes more sense to have pipe bombs need ignition to match other systems in the game (like welder bombs). 

Resolves #26239

https://github.com/space-wizards/space-station-14/issues/26239

## Technical details

I changed the pipebomb component from an OnUse, to a new component that looks at if an item is being used on it, that item must have an IgnitionSourceComponent and be set to ignited.

## Media

- [X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Pipe bombs now must be lit before exploding by something like a welder or lighter.

### CODE REVIEW

**Comment by Dutch-VanDerLinde** [comment]
can u add changelog and media

**Comment by Hmeister-real** [comment]
nice <3

**Comment by ChaseFlorom** [comment]
What media would be helpful here? I could have a screenshot of a pipe bomb in one hand and a welder in another, but is that helpful? I thought I had the changelog but it isn't showing, what do I need to do?

image

**Comment by Tayrtahn** [comment]
Move out outside of the comment block

**Comment by Dutch-VanDerLinde** [comment]
also put:

resolves [issue number] or fix [issue number] in front of the https://github.com/space-wizards/space-station-14/issues/26239 so it closes the issue if this PR is merged

**Comment by ChaseFlorom** [comment]
Alright, how are we looking now? Thanks for your help.

File: Content.Server/Explosion/EntitySystems/TriggerTimerOnIgniteSystem.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+using System.Threading.Tasks;
+using Content.Server.Explosion.EntitySystems;
+using Content.Server.Explosion.Components;
+using Robust.Shared.Random;
+using Content.Shared.Trigger;
+using Content.Server.IgnitionSource;
+
+
+namespace Content.Server.Explosion
+{
```
Comment (metalgearsloth) [inline Content.Server/Explosion/EntitySystems/TriggerTimerOnIgniteSystem.cs:21]: File-scoped namespaces please.

File: Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        ///     Allows changing the start-on-stick quality.
+        /// </summary>
+        [DataField("canToggleStartOnStick")] public bool AllowToggleStartOnStick;
+
+        /// <summary>
+        ///     Whether you can examine the item to see its timer or not.
+        /// </summary>
+        [DataField] public bool Examinable = true;
+    }
+}
```
Comment (metalgearsloth) [inline Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs:54]: This looks copy-pasted from OnuseTimerTrigger. Components should be made reuseable.

This should just be a generic "TriggerOnIgniteComponent" instead with all these copied options removed.

File: Content.Server/Explosion/EntitySystems/TriggerTimerOnIgniteSystem.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+            if (!TryComp<OnIgniteTimerTriggerComponent>(ent, out var timerTriggerComp))
+                return;
+
+            timerTriggerComp.Delay = _random.NextFloat(comp.Min, comp.Max);
+        }
+
+
+    }
+}
```
Comment (metalgearsloth) [inline Content.Server/Explosion/EntitySystems/TriggerTimerOnIgniteSystem.cs:82]: Same here on copy-paste.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs (hunk 2 above)
Comment (ChaseFlorom) [reply Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs:54]: I could probably remove some of these options, but why not leave some of them like beepinterval and beepsound? Wouldn't this allow for more items to use this component in the future? Same for below, IO could probably remove a bit, but why not leave some things the same if they work?

Comment (ChaseFlorom) [reply Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs:54]: @metalgearsloth Can I get your thoughts on this?

Comment (metalgearsloth) [reply Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs:54]: They should be on 1 component not spread across multiple, it just increases technical debt and makes it more of a pain to maintain the code.

Comment (deltanedas) [reply Content.Shared/Explosion/Components/OnIgniteTimerTriggerComponent.cs:54]: just have AutomaticTriggerComponent or whatever i called it then call TriggerSystem's public api to start the timer once its igniter

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**